src/extensibility/ExtensionManagerDialog.js

```diff
@@ -107,57 +110,98 @@ define(function (require, exports, module) {
                     .text(Strings.PROCESSING_EXTENSIONS)
                     .append("<span class='spinner inline spin'/>");
                 
-                ExtensionManager.removeMarkedExtensions()
+                var removeExtensionsPromise,
+                    updateExtensionsPromise,
+                    disableExtensionsPromise,
+                    removeErrors,
+                    updateErrors,
+                    disableErrors;
+                
+                removeExtensionsPromise = ExtensionManager.removeMarkedExtensions();
+                removeExtensionsPromise
+                    .fail(function (errorArray) {
+                        removeErrors = errorArray;
```

Comment:
There's no need to create `dialogs` at this scope. The error condition is processed in the `.fail` of the consolidated promise (`Async.waitForAll`).